### PR TITLE
docs(rest components): explain constraint validation

### PR DESCRIPTION
**Pull request by vursen**
Give a general description of how the constraint validation works, including such aspects as validation triggers, constraints, error messages, Binder, manual validation, for the following components:

- ComboBox
- MultiSelectComboBox
- Select
- RadioButtonGroup
- CheckboxGroup
- Checkbox

Part of https://github.com/vaadin/flow-components/issues/6545

### CODE REVIEW

File: vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java [hunk 1]
```diff
@@ -84,6 +85,31 @@
...
+ * <h2>Validation</h2>
+ * <p>
+ * CheckboxGroup comes with a built-in validation mechanism that verifies that
+ * at least one checkbox is selected when
+ * {@link #setRequiredIndicatorVisible(boolean) required} is enabled.
+ * <p>
+ * Validation is triggered whenever the user initiates a value change by
+ * toggling a checkbox. Programmatic value changes trigger validation as well.
+ * If validation fails, the component is marked as invalid and an error message
+ * is displayed below the input.
```
Comment (web-padawan) [inline vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java:97]: There is no single "input" in the group so maybe we could rephrase this:

```suggestion
 * is displayed below the group.
```

File: vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBoxBase.java [hunk 2]
```diff
@@ -355,20 +355,46 @@ protected void setFilter(String filter) {
...
+    @Override
+    public boolean isRequiredIndicatorVisible() {
+        return super.isRequiredIndicatorVisible();
+    }
+
+    /**
+     * Alias for {@link #isRequiredIndicatorVisible()}
      *
-     * @return {@code true} if the component requires a value to be valid
+     * @return true if the field is required, false otherwise
```
Comment (web-padawan) [inline vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBoxBase.java:387]: We probably could keep using `{@code true}` for formatting in the IDE. Not a blocker though.

File: vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/RadioButtonGroup.java [hunk 3]
```diff
@@ -71,6 +72,32 @@
...
+ * <p>
+ * Radio Button Group comes with a built-in validation mechanism that verifies
+ * that a radio button is selected when
+ * {@link #setRequiredIndicatorVisible(boolean) required} is enabled. Validation
+ * is triggered whenever the user selects a radio button or the value is updated
+ * programmatically. In practice, however, the required error can only occur if
+ * the value is cleared programmatically. This is because radio buttons, by
+ * design, don't allow users to clear a selection through UI interaction. If the
+ * required error occurs, the component is marked as invalid and an error
+ * message is displayed below the input.
```
Comment (web-padawan) [inline vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/RadioButtonGroup.java:85]: Same as for checkbox group, I'd not use "input" here.

```suggestion
 * message is displayed below the group.
```

File: vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/RadioButtonGroup.java [hunk 4]
```diff
@@ -71,6 +72,32 @@
...
+ * <p>
+ * The required error message can be configured using either
+ * {@link SelectI18n#setRequiredErrorMessage(String)} or
```
Comment (web-padawan) [inline vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/RadioButtonGroup.java:88]:
```suggestion
 * {@link RadioButtonGroupI18n#setRequiredErrorMessage(String)} or
```

File: vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java (hunk 1 above)
Comment (vursen) [reply vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java]: Good catch, thanks, fixed.

File: vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBoxBase.java (hunk 2 above)
Comment (vursen) [reply vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBoxBase.java]: Fixed

File: vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/RadioButtonGroup.java (hunk 3 above)
Comment (vursen) [reply vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/RadioButtonGroup.java]: Fixed.

File: vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/RadioButtonGroup.java (hunk 4 above)
Comment (vursen) [reply vaadin-radio-button-flow-parent/vaadin-radio-button-flow/src/main/java/com/vaadin/flow/component/radiobutton/RadioButtonGroup.java]: Fixed.

**Review by web-padawan — APPROVED** [review]

**Review by rolfsmeds — APPROVED** [review]

File: vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/Checkbox.java [hunk 5]
```diff
@@ -212,6 +237,35 @@ protected void onAttach(AttachEvent attachEvent) {
         ClientValidationUtil.preventWebComponentFromModifyingInvalidState(this);
     }
 
+    /**
+     * Sets whether the user is required to select the checkbox. When required,
+     * an indicator appears next to the label and the field invalidates if the
+     * checkbox is deselected.
```
Comment (vursen) [inline vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/Checkbox.java:243]:
```suggestion
     * checkbox is first selected and then deselected.
```

**Pull request merged**